### PR TITLE
support adjusting the grayscale calculation. It used to calculate the…

### PR DESCRIPTION
**Pull request by jongleur1983**
… gray value by constant quota 3 x red + 4 x green + 1 x blue / 8. With this change it's possible to adjust that with additional options. This allows to change the weight between the color channels on the one hand, and enlighten/darken the resulting image (lighter: quotaDivider > quotaRed + quotaGreen + quotaBlue)

### CODE REVIEW

File: TileLayer.Grayscale.js [hunk 1]
```diff
@@ -4,7 +4,11 @@
 
 L.TileLayer.Grayscale = L.TileLayer.extend({
 	options: {
-		enableCanvas: true
+		enableCanvas: true,
+		quotaRed: 3,
+		quotaGreen: 4,
+		quotaBlue: 1,
+		quotaDivider: 3+4+1,
```
Comment (Zverik) [inline TileLayer.Grayscale.js]: Why not use `quotaRed + quotaGreen + quotaBlue` in the equation?

**Comment by jongleur1983** [comment]
well... good point... fixed

**Comment by Zverik** [comment]
No, I meant, not make this configurable at all -- just use the sum in the equation in line 42. Modifying it doesn't make much sense.

**Comment by jongleur1983** [comment]
of course it does. that's what allows to lighten or darken the map.
In particular I use this to get a lighter map as a result:

    L.tileLayer.grayscale('http://tile.openstreetmap.org/{z}/{x}/{y}.png', {
            attribution: 'Map data &copy; <a href="http://openstreetmap.org/">OpenStreetMap</a> contributors',
            maxZoom: 18, minZoom: 2,
			quotaDivider: 13,
			quotaRed: 6,
			quotaGreen: 7,
			quotaBlue: 2			
        }).addTo(map);

Your're right: it's not as nice as the default, but it better matches my need in the context of the target website, as the default result (taking osm-mapnik as a source layer) is too dark.

You see, that the sum would be 6+7+2=15, but the quotaDivider I use is 13, so yes, this freedom of configuration is exactly what I want to achieve here.

**Comment by Zverik** [comment]
Okay, thank you then :)

**Pull request merged**